pkg/cloudprovider/aws/instance.go

```diff
@@ -143,6 +197,13 @@ func (p *InstanceProvider) launchInstances(ctx context.Context, constraints *v1a
 	if err != nil {
 		return nil, fmt.Errorf("creating fleet %w", err)
 	}
+	if len(createFleetOutput.Errors) > 0 {
+		for _, fleetError := range createFleetOutput.Errors {
```

Comment:
```suggestion
		for _, err := range createFleetOutput.Errors {
```
In go, it's common to shadow `err` in nested scopes.